app/uk/gov/hmrc/platformstatusbackend/services/LoadGenerator.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.platformstatusbackend.services
+
+import play.api.Logger
+
+import java.util.UUID
+import java.util.concurrent.{LinkedBlockingQueue, ThreadPoolExecutor}
+import javax.inject.Singleton
+import scala.concurrent.duration.{Duration, SECONDS}
+import scala.util.Random
+
+case class TestAccepted(id: UUID)
+case class TestInProgress()
+
+@Singleton()
+object LoadGenerator {
+
+  private val logger = Logger(this.getClass)
+
+  val threadPool = new ThreadPoolExecutor(1, 16, 0L, SECONDS, new LinkedBlockingQueue[Runnable]());
+
+  sys.addShutdownHook {
+    threadPool.shutdownNow()
+  }
+
+  def cpuLoad(threadCount: Int, duration: Duration): Either[TestInProgress, TestAccepted] = {
+    runTest( (testId: TestAccepted, threadId: Int) => () => {
+      logger.info(s"Test ${testId.id} Thread $threadId - maxing cpu for ${duration.toSeconds} seconds")
+      val endAt = System.currentTimeMillis() + duration.toMillis
+      while (System.currentTimeMillis() < endAt) {}
+      logger.info(s"Test ${testId.id} Thread $threadId - completed maxing cpu")
+    }, threadCount)
+  }
+
+  def cpuTasks(threadCount: Int, tasks: Int) : Either[TestInProgress, TestAccepted] = {
+    runTest( (test:TestAccepted, threadId: Int) => () => {
+
+      val start = System.currentTimeMillis()
+      var max = 0
+      val tasksPerThread = tasks / threadCount
+
+      logger.info(s"Test ${test.id} Thread $threadId - processing $tasksPerThread tasks")
+
+      Range(0, tasksPerThread).foreach { _ =>
+        val data = Iterator.continually(Random.nextInt()).take(4096).toList
+        val res = data.foldLeft(0)( (acc, i) => acc + { data.foldLeft(0)((acc2, i2) => acc2 + i2*i) })
+        max = Math.max(res, max) // might not be needed, assignment here was to ensure the compiler doesn't optimize away the line above
+      }
+
+      logger.info(s"Test ${test.id} Thread $threadId - complete $tasksPerThread tasks in ${System.currentTimeMillis() - start} ms")
+
+    }, threadCount)
+  }
+
+  private def runTest( f : (TestAccepted, Int) => Runnable, threadCount: Int ): Either[TestInProgress, TestAccepted] = {
+    if(threadPool.getActiveCount > 0) {
```

Comment:
should this be `synchronized`? we could get a race condition where two requests both get through and submit tasks?